apps/oc_chef_wm/itest/oc_chef_wm_keys_SUITE.erl

```diff
@@ -92,145 +93,239 @@ all() ->
+    ?assertMatch({ok, "404", _, _}, Result).
+
+put_rename_client_key(_Config) ->
+    validate_rename(client , "201", "404").
+
+put_rename_duplicate_client_key(_Config) ->
+    validate_rename(client, "409", "200").
+
+put_rename_user_key(_Config) ->
+    validate_rename(user, "201", "404").
+
+put_rename_duplicate_user_key(_Config) ->
+    validate_rename(user, "409", "200").
+
+put_full_client_key(Config) ->
+    validate_put_full_key(Config, client).
+
+put_full_user_key(Config) ->
+    validate_put_full_key(Config, user).
+
+put_valid_partial_user_key(Config) ->
+    validate_put_partial_key_valid(Config, user, <<"expiration_date">>, ?KEY2EXPIRESTRING),
+    validate_put_partial_key_valid(Config, user, <<"public_key">>, proplists:get_value(alt_pubkey, Config)).
+
+put_valid_partial_client_key(Config) ->
+    validate_put_partial_key_valid(Config, client, <<"expiration_date">>, ?KEY2EXPIRESTRING),
+    validate_put_partial_key_valid(Config, client, <<"public_key">>, proplists:get_value(alt_pubkey, Config)).
+
+put_invalid_partial_user_key(_Config) ->
+    validate_put_partial_key_invalid(user, <<"name">>, <<"bob^was^here">>),
+    validate_put_partial_key_invalid(user, <<"expiration_date">>, <<"yesterday">>),
+    validate_put_partial_key_invalid(user, <<"public_key">>, <<"bad key">>).
+
+put_invalid_partial_client_key(_Config) ->
+    validate_put_partial_key_invalid(client, <<"name">>, <<"bob^was^here">>),
+    validate_put_partial_key_invalid(client, <<"expiration_date">>, <<"yesterday">>),
+    validate_put_partial_key_invalid(client, <<"public_key">>, <<"bad key">>).
+
+% TODO - authing requestor
+% I'm leaving this commented out but present, because I'd like to find a way to make it work.
+% Currently, our stubbed chef_authn mock can't populate a requestor id, which means
+% our check to see if keyname + requestor id matches current auth will always pass without
+% matching, because required id is never populated.
+% I think we can do this by making our mock chef_authn a stateful process (gen_server would be simplest)
+% which would allow us to set the requestor id to mock for each reque We could also then extend it
+% to perform better authz behavior validation throughout our tests
+% We'll want to add similar tests for PUT.
+%delete_authing_client_key(_Config) ->
+    %Result = http_named_key_request(delete, client, ?CLIENT_NAME, "default"),
+    %?assertMatch({ok, "403", _, _}, Result).
+
+%delete_authing_client_key_with_different_user(_Config) ->
+    %% This is specifically intended to validate that we can use a different user's
+    %% "default" key to authenticate, and delete the client's "default" key without
+    %% failing simply because the key name match(The stubbed authz check
+    %% will match on key 'default')
+    %Result = http_named_key_request(delete, client, ?ADMIN_USER_NAME, "default"),
+    %?assertMatch({ok, "200", _, _}, Result).
```

Comment:
:clap: I would love to get auth tests working here also. At least we are documenting that we want to / don't have them yet.